### PR TITLE
feat: add "approve in wallet" prompt on chain selector

### PR DESCRIPTION
**Pull request by vm**
Screen Shot 2022-09-20 at 2 16 49 PM
Screen Shot 2022-09-20 at 2 16 11 PM
Screen Shot 2022-09-20 at 2 16 16 PM

### CODE REVIEW

File: src/components/NavBar/ChainSelector.tsx [hunk 1]
```diff
@@ -74,6 +44,18 @@ export const ChainSwitcher = ({ leftAlign }: ChainSwitcherProps) => {
...
   useSyncChainQuery()
 
+  const [pendingChainId, setPendingChainId] = useState<SupportedChainId | undefined>(undefined)
+
+  const onSelectChain = useCallback(
+    async (targetChainId: SupportedChainId) => {
+      setPendingChainId(targetChainId)
+      await selectChain(targetChainId)
+      setPendingChainId(undefined)
+      toggleOpen()
```
Comment (cbachmeier) [inline src/components/NavBar/ChainSelector.tsx:54]: Can we add a check here to make sure the Selector is open before toggling? Currently if I close the modal before approving in my wallet, the modal reopens.

File: src/components/NavBar/ChainSelectorRow.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  text-align: left;
+  line-height: 24px;
+  border: none;
+  justify-content: space-between;
+  padding: 10px 8px;
+  margin: 0 8px;
+  cursor: pointer;
+  border-radius: 12px;
+  color: ${({ theme }) => theme.textPrimary};
+  width: 240px;
```
Comment (cbachmeier) [inline src/components/NavBar/ChainSelectorRow.tsx:24]: Width is off when left aligned
Screen Shot 2022-09-20 at 11 30 08

Comment (cbachmeier) [reply src/components/NavBar/ChainSelectorRow.tsx:24]: Might actually be right padding
Screen Shot 2022-09-20 at 11 31 28

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/components/NavBar/ChainSelector.tsx (hunk 1 above)
Comment (vm) [reply src/components/NavBar/ChainSelector.tsx]: ended up changing out of the reducer to solve it

**Review by cbachmeier — APPROVED** [review]
LGTM

**Pull request merged**